test/unit/lib/serverless.test.js

```diff
@@ -174,4 +179,61 @@ describe('test/unit/lib/serverless.test.js', () => {
       expect(serverless.config).to.have.property('servicePath');
     });
   });
+
+  describe('Extend configuration', () => {
+    let awsRegion;
+    const pluginConfig =
+      require('../../fixtures/programmatic/plugin/extend-config-plugin').pluginConfig;
+
+    const serverlessPath = path.resolve(__dirname, '../../../scripts/serverless.js');
+
+    before(() => {
+      awsRegion = process.env.AWS_REGION;
+      process.env.AWS_REGION = 'us-east-1';
+    });
+
+    after(() => {
+      process.env.AWS_REGION = awsRegion;
+    });
+
+    it('Extends configuration with given values', async () => {
+      const customExt = { custom: {} };
+      const configExt = {
+        plugins: ['./extend-config-plugin/index.js'],
+        provider: {
+          stage: 'dev',
+        },
+        custom: {},
+      };
+      _.set(customExt, pluginConfig.overwriteValuePath, 'test_value');
+
+      const { servicePath: serviceDir } = await programmaticFixturesEngine.setup('plugin', {
+        configExt,
+      });
+      let configuration;
+      try {
+        const serverlessProcess = await spawn('node', [serverlessPath, 'print'], {
+          cwd: serviceDir,
+        });
+        configuration = yaml.load(String(serverlessProcess.stdoutBuffer));
+      } catch (error) {
```

Comment:
Sure. This was just to show why it failed. Now the error will be "exited with code 1".

Reply:
> Sure. This was just to show why it failed. Now the error will be "exited with code 1".

With debug logging (you can enable with e.g. `LOG_LEVEL=debug`) you'll get all the underlying `stdout` etc. In CI in case of failure we dump all such logs. So we should be good